todo-react/src/components/Card.tsx

```diff
@@ -0,0 +1,167 @@
+  transition: all 0.2s ease-in-out;
+`;
+
+const TitleContainer = styled.div``;
+
+const CreatedAt = styled.span`
+  font-size: 0.8rem;
+  color: inherit;
+  opacity: 0.5;
+`;
+
+const ControlContainer = styled.div<{ done: boolean }>`
+  display: flex;
+  justify-content: flex-end;
+  align-items: center;
+  gap: 0.5rem;
+  svg {
+    transition: all 0.2s ease-in-out;
+    color: inherit;
+    cursor: pointer;
+    font-size: 2.5rem;
+    padding: 0.5rem;
+    border-radius: 9999px;
+    :hover {
+      background-color: ${({ done }) => (done ? "#fff" : "#2c3e50")};
+      color: ${({ done }) => (done ? "#2c3e50" : "#fff")};
+    }
+    :last-child:hover {
+      background-color: #e74c3c;
+      color: #fff;
+    }
+  }
+`;
+
+const NewTodoForm = styled.form``;
+
+const NewTodoInput = styled.input.attrs({
+  type: "text",
+  required: true,
+  maxLength: 20,
+})`
+  all: unset;
+`;
+
+const Title = styled.h3``;
+
+interface CardProps extends Todo {
+  handleEditTodo: (id: number, newTitle: string) => void;
+  handleDeleteTodo: (id: number) => void;
+  handleCompleteTodo: (id: number) => void;
+}
+
+const Card = ({
+  title,
+  createdAt,
+  id,
+  completed,
+  handleEditTodo,
+  handleDeleteTodo,
+  handleCompleteTodo,
+}: CardProps) => {
+  const [isEditing, setIsEditing] = useState(false);
+  const [newTitle, setNewTitle] = useState(title);
+  const inputRef = useRef<HTMLInputElement>(null);
+
+  const editTitle = ({
+    currentTarget: { value },
+  }: React.ChangeEvent<HTMLInputElement>) => {
+    setNewTitle(value);
+  };
+
+  const toggleEdit = useCallback(() => {
+    if (newTitle !== title) setNewTitle(title);
+    setIsEditing(!isEditing);
+  }, [isEditing, newTitle, title]);
+
+  const submitEditedTitle = useCallback(
+    (e?: React.FormEvent<HTMLFormElement>) => {
+      e?.preventDefault();
+      handleEditTodo(id, newTitle);
+      toggleEdit();
+    },
+    [handleEditTodo, id, newTitle, toggleEdit]
+  );
+
+  const todoContent = useMemo(
+    () =>
+      isEditing ? (
+        <NewTodoForm onSubmit={submitEditedTitle}>
+          <NewTodoInput ref={inputRef} value={newTitle} onChange={editTitle} />
+        </NewTodoForm>
+      ) : (
+        <Title>{title}</Title>
+      ),
+    [isEditing, newTitle, submitEditedTitle, title]
+  );
+
+  const controllers = useMemo(
+    () =>
+      isEditing ? (
+        <>
+          <MdOutlineCheckCircleOutline onClick={() => submitEditedTitle()} />
+          <MdHighlightOff onClick={toggleEdit} />
+        </>
+      ) : (
+        <>
+          <MdEdit onClick={toggleEdit} />
+          <MdDone onClick={() => handleCompleteTodo(id)} />
+          <MdClose onClick={() => handleDeleteTodo(id)} />
+        </>
+      ),
+    [
+      isEditing,
+      submitEditedTitle,
+      toggleEdit,
+      handleCompleteTodo,
+      id,
+      handleDeleteTodo,
+    ]
+  );
+
+  useEffect(() => {
+    if (isEditing) {
+      inputRef.current?.focus();
+    }
+  }, [isEditing, inputRef]);
+
```

Comment:
```suggestion
  useEffect(() => {
    if (isEditing && inputRef.current) {
      inputRef.current.focus();
    }
  }, [isEditing]);
```
이렇게 표현할 수 있답니다!
ref에 대한 의존은 대부분 불필요하답니다.